parcels/kernel.py

```diff
@@ -219,7 +219,7 @@ def load_lib(self):
     def execute_jit(self, pset, endtime, dt):
         """Invokes JIT engine to perform the core update loop"""
         if len(pset) > 0 and pset.particle_data['xi'].ndim == 2:
-            assert pset.fieldset.gridset.size == pset.particle_data['xi'].shape[0], \
+            assert pset.fieldset.gridset.size == pset.particle_data['xi'].shape[1], \
```

Comment:
great - so now the order is again: particles['xi'][<particle_index>][<grid_index>] -> I like that a lot!